Angular/src/app/calculator/calculator.component.spec.ts

```diff
@@ -1,24 +1,56 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
-
+import {ComponentFixture, TestBed, waitForAsync} from '@angular/core/testing';
 import { CalculatorComponent } from './calculator.component';
 import {ReactiveFormsModule} from "@angular/forms";
 import {SharedModule} from "../../shared/shared.module";
+import {CalculatorService} from "./services/calculator.service";
+import {DomTestingService} from "../helpers/dom-testing.service";
+import {provideMock, Spied} from "../helpers/provide-mock";
+
+
+const dataTests = {
+  title: '[data-test="calculator-title"]',
+  firstNumberInput: '[data-test="first-number-input"]',
+  secondNumberInput: '[data-test="second-number-input"]',
+  calculateButton: '[data-test="calculate-button"]',
+  result: '[data-test="result"]',
+};
 
 describe('CalculatorComponent', () => {
   let component: CalculatorComponent;
   let fixture: ComponentFixture<CalculatorComponent>;
+  let domTestingService: DomTestingService;
+  let calculatorServiceMock: Spied<CalculatorService>;
 
-  beforeEach(() => {
+  beforeEach(waitForAsync(() => {
     TestBed.configureTestingModule({
       declarations: [CalculatorComponent],
+      providers: [provideMock(CalculatorService)],
       imports: [SharedModule, ReactiveFormsModule],
-    });
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
     fixture = TestBed.createComponent(CalculatorComponent);
+    domTestingService = new DomTestingService(fixture);
     component = fixture.componentInstance;
-    fixture.detectChanges();
+    calculatorServiceMock = TestBed.inject(CalculatorService);
   });
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
+  describe('Component finished rendering', () => {
+    beforeEach(() => {
+      fixture.detectChanges();
+    });
+
+    it('should show title', () => {
+      expect(domTestingService.getElement(dataTests.title)).toBeTruthy();
+    });
+
+    it('first number input should be 0', () => {
```

Comment:
can you change it to `"first number input should be the default"`,
inside the test, create variable `expectedDefaultNumber` and use it in the expectations,
same for the other tests